sycl/include/sycl/ext/oneapi/experimental/builtins.hpp

```diff
@@ -59,8 +59,8 @@ namespace experimental {
 // guarded using __SYCL_DEVICE_ONLY__ preprocessor macro or avoided in favor
 // of more portable solutions if needed
 //
-template <typename... Args>
-int printf(const __SYCL_CONSTANT_AS char *__format, Args... args) {
+template <typename FormatT, typename... Args>
+int printf(const FormatT *__format, Args... args) {
```

Comment:
There is a comment above mentioning address space restriction - I suggest we update it to say that in simple cases (like direct use of a literal within a printf call) compiler is able to automatically perform address space conversion, but in more complex cases users still have to perform tricks to explicitly put literal into constant address space